include/envoy/http/filter.h

```diff
@@ -36,6 +36,15 @@ enum class FilterHeadersStatus {
   // results in creating a header only request/response.
   // This status MUST NOT be returned by decodeHeaders() when end_stream is set to true.
   ContinueAndEndStream,
+  // Continue iteration to remaining filters, but do not end the stream.
+  //
+  // Used when a filter wants to add a body to headers-only request/response, but this body is not
+  // readily available.
+  // This causes the headers iteration to continue, but the stream does not end.
+  // The filter is responsible to continue the stream by providing a body through
+  // injectEncodedDataToFilterChain()/injectDecodedDataToFilterChain().
+  // If the filter cannot provide a body the stream should be reset.
+  ContinueAndDontEndStream,
```

Comment:
The current implementation supports adding a body only through `inject...`. Adding a body through `addEncodedData`/`addDecodedData` + `continueEncoding`/`continueDecoding` can be supported be adding a new `IterationState`, for e.g. `IterationState::ContinueAndDontEndStream` or `IterationState::WaitingForBody`.

Currently the asserts in `FilterManager::commonContinue()` and `FilterManager::allowIteration()` prevent this, but I tried commenting the asserts and it should work. So all we need to do to support this is to add a new `IterationState`. Do you think we should do it now? or add a TODO to do it if a filter needs it in the future?

Also, can you confirm the correctness of this line ` If the filter cannot provide a body the stream should be reset.`?

Reply:
Of course I'll add another unit test for this if you see that we should support it.

Reply:
I think just supporting inject for now is OK, but please add a bunch of comments/TODOs and make sure it fails in some easily understood way via ASSERT, etc.

> Also, can you confirm the correctness of this line If the filter cannot provide a body the stream should be reset.?

Yes I think this is correct.